lib/AST/Type.cpp

```diff
@@ -6021,8 +6021,7 @@ TypeBase::getAutoDiffTangentSpace(LookupConformanceFn lookupConformance) {
       newElts.push_back(elt.getWithType(eltSpace->getType()));
     }
     if (newElts.empty())
-      return cache(
-          TangentSpace::getTuple(ctx.TheEmptyTupleType->castTo<TupleType>()));
```

Comment:
We may want to add a "non-empty tuple" assertion to `TangentSpace::getTuple` to prevent this from happening elsewhere.

Reply:
That sounds like a good idea. I’ll change that before this PR gets merged.

Reply:
Fixed.